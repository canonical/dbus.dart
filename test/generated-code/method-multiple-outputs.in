<node>
  <interface name="com.example.Test">
    <method name="Hello">
      <arg name="byte_value" type="y" direction="out"/>
      <arg name="boolean_value" type="b" direction="out"/>
      <arg name="int16_value" type="n" direction="out"/>
      <arg name="uint16_value" type="q" direction="out"/>
      <arg name="int32_value" type="i" direction="out"/>
      <arg name="uint32_value" type="u" direction="out"/>
      <arg name="int64_value" type="x" direction="out"/>
      <arg name="uint64_value" type="t" direction="out"/>
      <arg name="double_value" type="d" direction="out"/>
      <arg name="string_value" type="s" direction="out"/>
      <arg name="object_path_value" type="o" direction="out"/>
      <arg name="signature_value" type="g" direction="out"/>
      <arg name="variant_value" type="v" direction="out"/>
      <arg name="unix_fd_value" type="h" direction="out"/>
      <arg name="struct_value" type="(si)" direction="out"/>
      <arg name="array_value_y" type="ay" direction="out"/>
      <arg name="array_value_b" type="ab" direction="out"/>
      <arg name="array_value_n" type="an" direction="out"/>
      <arg name="array_value_q" type="aq" direction="out"/>
      <arg name="array_value_u" type="ai" direction="out"/>
      <arg name="array_value_n" type="au" direction="out"/>
      <arg name="array_value_x" type="ax" direction="out"/>
      <arg name="array_value_t" type="at" direction="out"/>
      <arg name="array_value_d" type="ad" direction="out"/>
      <arg name="array_value_s" type="as" direction="out"/>
      <arg name="array_value_o" type="ao" direction="out"/>
      <arg name="array_value_g" type="ag" direction="out"/>
      <arg name="array_value_v" type="av" direction="out"/>
      <arg name="array_value_h" type="ah" direction="out"/>
      <arg name="array_value_struct" type="a(si)" direction="out"/>
      <arg name="array_value_array" type="aas" direction="out"/>
      <arg name="dict_value_sv" type="a{sv}" direction="out"/>
      <arg name="dict_value_us" type="a{us}" direction="out"/>
      <arg name="dict_value_dict" type="a{sa{sv}}" direction="out"/>
    </method>
  </interface>
</node>
