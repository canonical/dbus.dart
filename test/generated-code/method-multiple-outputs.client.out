import 'dart:io';
import 'package:dbus/dbus.dart';

class ComExampleTest extends DBusRemoteObject {
  ComExampleTest(DBusClient client, String destination, DBusObjectPath path) : super(client, name: destination, path: path);

  /// Invokes com.example.Test.Hello()
  Future<List<DBusValue>> callHello({bool noAutoStart = false, bool allowInteractiveAuthorization = false}) async {
    var result = await callMethod('com.example.Test', 'Hello', [], replySignature: DBusSignature('ybnqiuxtdsogvh(si)ayabanaqaiauaxatadasaoagavaha(si)aasa{sv}a{us}a{sa{sv}}'), noAutoStart: noAutoStart, allowInteractiveAuthorization: allowInteractiveAuthorization);
    return result.returnValues;
  }
}
